app/src/main/java/co/icanteach/projectx/MainActivity.kt

```diff
@@ -46,10 +45,10 @@ class MainActivity : AppCompatActivity() {
             renderStatusResult(contents)
         }
 
-        savedInstanceState.runIfNull {
-            fetchMovies(FIRST_PAGE)
-        }
         initPopularTVShowsRecyclerView()
+
+        savedInstanceState ?: fetchMovies(FIRST_PAGE)
```

Comment:
For fragments, unfortunately we should do this but when we use VM, we should avoid doing this.
this part should be moved to VM `init` method.

```kotlin
init {
        fetchMovies(FIRST_PAGE)
}
```